internal/providers/terraform/terragrunt_hcl_provider.go

```diff
@@ -575,6 +574,13 @@ func (p *TerragruntHCLProvider) runTerragrunt(opts *tgoptions.TerragruntOptions)
 		}
 	}
 
+	for _, config := range terragruntConfig.GenerateConfigs {
```

Comment:
@hugorut do we need to wrap this in the terragruntSourceLock?

Reply:
yes, because it manipulates the `workingDir`